### PR TITLE
Feature/data table linker updates

### PR DESCRIPTION
**Pull request by makosblade**

### CODE REVIEW

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt [hunk 1]
```diff
@@ -305,7 +314,9 @@ class PostgresLinkingQueryService(private val hds: HikariDataSource) : LinkingQu
                     val ps = connection.prepareStatement(ENTITY_KEY_IDS_OF_LINKING_IDS_SQL)
                     val linkingIdsArray = PostgresArrays.createUuidArray(connection, linkingIds)
 
-                    ps.setObject(1, linkingIdsArray)
+                    val partitions = PostgresArrays.createIntArray( connection, getPartitionsInfo( linkingIds, partitionManager.getAllPartitions() ))
+                    ps.setArray(1, partitions)
+                    ps.setObject(2, linkingIdsArray)
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: This should be probably setArray

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt [hunk 2]
```diff
@@ -404,8 +429,8 @@ private val LINKABLE_ENTITY_SET_IDS = "SELECT ${ID.name} " +
 
 private val LOCK_CLUSTERS_SQL = "SELECT 1 FROM ${MATCHED_ENTITIES.name} WHERE ${LINKING_ID.name} = ? FOR UPDATE"
 
-private val ENTITY_KEY_IDS_OF_LINKING_IDS_SQL = "SELECT ${LINKING_ID.name}, array_agg(${ID.name}) AS ${ENTITY_KEY_IDS.name} " +
+private val ENTITY_KEY_IDS_OF_LINKING_IDS_SQL = "SELECT ${LINKING_ID.name}, array_agg(${ID.name}) AS ${ENTITY_KEY_IDS_COL.name} " +
         "FROM ${IDS.name} " +
-        "WHERE ${VERSION.name} > 0 AND ${LINKING_ID.name} IS NOT NULL AND  ${LINKING_ID.name} = ANY( ? ) " +
+        "WHERE ${PARTITION.name} = ? AND ${VERSION.name} > 0 AND ${LINKING_ID.name} IS NOT NULL AND ${LINKING_ID.name} = ANY( ? ) " +
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: I think this should be `${PARTITION.name} = ANY( ? )`

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt [hunk 3]
```diff
@@ -385,17 +409,18 @@ private val BLOCKS_BY_SIZE_SQL = "SELECT ${SRC_ENTITY_SET_ID.name} as entity_set
...
-        "WHERE ${ENTITY_SET_ID.name} = ANY(?) AND ${LAST_LINK.name} < ${LAST_WRITE.name} AND ( ${LAST_INDEX.name} >= ${LAST_WRITE.name}) AND ( ${LAST_INDEX.name} > '-infinity'::timestamptz) " +
+        "FROM ${IDS.name} " +
+        "WHERE ${PARTITION.name} = ANY(?) AND ${ENTITY_SET_ID.name} = ANY(?) AND ${LAST_LINK.name} < ${LAST_WRITE.name} AND ( ${LAST_INDEX.name} >= ${LAST_WRITE.name}) AND ( ${LAST_INDEX.name} > '-infinity'::timestamptz) " +
         "AND ${VERSION.name} > 0 LIMIT ?"
 
 private val ENTITY_KEY_IDS_NOT_LINKED = "SELECT ${ENTITY_SET_ID.name},${ID.name} " +
-        "FROM ${ENTITY_KEY_IDS.name} " +
-        "WHERE ${ENTITY_SET_ID.name} = ANY(?) AND ${LAST_LINK.name} < ${LAST_WRITE.name} " +
+        "FROM ${IDS.name} " +
+        "WHERE ${PARTITION.name} = ? AND ${ENTITY_SET_ID.name} = ANY(?) AND ${LAST_LINK.name} < ${LAST_WRITE.name} " +
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: This should be probably also ` = ANY( ? )`

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt (hunk 1 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: Ah, nice catch!

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt (hunk 2 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: Willfix!

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt (hunk 3 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: Derp

File: src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt [hunk 4]
```diff
@@ -87,8 +90,10 @@ class PostgresLinkingQueryService(private val hds: HikariDataSource) : LinkingQu
             val connection = hds.connection
             val ps = connection.prepareStatement(ENTITY_KEY_IDS_NEEDING_LINKING)
             val arr = PostgresArrays.createUuidArray(connection, entitySetIds)
-            ps.setObject(1, arr)
-            ps.setObject(2, limit)
+            val partitions = getPartitionsAsPGArray(connection, entitySetIds)
+            ps.setArray(1, partitions)
+            ps.setObject(2, arr)
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: And this one too (setArray)

Comment (makosblade) [reply src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: 👍 I found a couple of others too 🤦‍♂

Comment (you-leee) [reply src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt]: 🙌

**Pull request merged**